application/actions/requests.go

```diff
@@ -351,6 +351,50 @@ func requestsAddMeAsPotentialProvider(c buffalo.Context) error {
 	return c.Render(200, render.JSON(output))
 }
 
+// swagger:operation DELETE /requests/{request_id}/potentialprovider Requests RemoveMeAsPotentialProvider
+//
+// Removes the current user as a potential provider to the request
+//
+// ---
+// responses:
+//   '200':
+//     description: The id (uuid) of the request for which the potential provider was removed
+func requestsRemoveMeAsPotentialProvider(c buffalo.Context) error {
+	cUser := models.CurrentUser(c)
+	tx := models.Tx(c)
+
+	id, err := getUUIDFromParam(c, "request_id")
+	if err != nil {
+		return reportError(c, err)
+	}
+	domain.NewExtra(c, "requestID", id)
+
+	var provider models.PotentialProvider
+	if err := provider.FindWithRequestUUIDAndUserUUID(tx, id.String(), cUser.UUID.String(), cUser); err != nil {
+		appError := api.NewAppError(err, api.ErrorRemoveMeAsPotentialProviderForbidden, api.CategoryForbidden)
+		if strings.Contains(err.Error(), "unable to find User") {
+			appError.Key = api.ErrorRemoveMeAsPotentialProviderFindUser
+		} else if strings.Contains(err.Error(), "unable to find Request") {
+			appError.Key = api.ErrorGetRequest
+			appError.Category = api.CategoryUser
+		} else if strings.Contains(err.Error(), "unable to find PotentialProvider") {
+			appError.Key = api.ErrorRemoveMeAsPotentialProviderFindProvider
+			appError.Category = api.CategoryUser
+		}
+
+		return reportError(c, appError)
+	}
+
+	domain.NewExtra(c, "request", id)
+
+	if err := provider.Destroy(tx); err != nil {
+		appError := api.NewAppError(err, api.ErrorRemoveMeAsPotentialProviderDestroyIt, api.CategoryInternal)
+		return reportError(c, appError)
+	}
+
+	return c.Render(200, render.String(id.String()))
```

Comment:
I think this can just return a 204 with no data.

Reply:
:+1: